packages/amplify_datastore/android/build.gradle

```diff
@@ -29,15 +29,37 @@ android {
 
     sourceSets {
         main.java.srcDirs += 'src/main/kotlin'
+        test.java.srcDirs += 'src/test/kotlin'
+        test.resources.srcDirs += '../test/resources'
     }
+
     defaultConfig {
         minSdkVersion 16
     }
     lintOptions {
         disable 'InvalidPackage'
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    testOptions {
+        unitTests {
+            includeAndroidResources = true
+        }
+    }
 }
 
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation "com.amplifyframework:aws-datastore:1.1.3"
+    implementation "com.amplifyframework:aws-api-appsync:1.1.3"
+    testImplementation 'junit:junit:4.13'
+    testImplementation 'org.mockito:mockito-core:3.1.0'
+    testImplementation 'org.mockito:mockito-inline:3.1.0'
+    testImplementation 'androidx.test:core:1.2.0'
+    testImplementation 'org.robolectric:robolectric:4.3.1'
+    testImplementation 'com.fasterxml.jackson.core:jackson-core:2.6.3'
+    testImplementation 'com.fasterxml.jackson.core:jackson-annotations:2.6.3'
+    testImplementation 'com.fasterxml.jackson.core:jackson-databind:2.6.3'
```

Comment:
Yeah I used Gson before, but it's a little naive in handling a JSON to a generic hashmap without given a proper type object. This is difficult when we are deserializing QueryPredicates and without the use of ModelSchema (JSONNumber(4) is deserialized as Double(4.0) by default by Gson). This is why I changed to Jackson for test implementation to avoid writing unncessary complex type objects for deserialization.